editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -1035,6 +1035,29 @@ void CanvasItemEditor::_viewport_input_event(const InputEvent &p_event) {
 
 		const InputEventMouseButton &b = p_event.mouse_button;
 
+		if (b.button_index == GESTURE_MAGNIFY) {
+
+			if (zoom < MIN_ZOOM) {
```

Comment:
I think I will try to add gesture and wheel events again, hope to get it done this time. And `direction_value` is a good name.